File: web/src/test/java/top/bettercode/summer/web/support/EmbeddedIdBean.java
package top.bettercode.summer.web.support;

import java.util.Objects;

public class EmbeddedIdBean {

  private String name;
  private Integer intCode;
  private Long price;

  public String getName() {
    return name;
  }

  public void setName(String name) {
    this.name = name;
  }

  public Integer getIntCode() {
    return intCode;
  }

  public void setIntCode(Integer intCode) {
    this.intCode = intCode;
  }

  public Long getPrice() {
    return price;
  }

  public void setPrice(Long price) {
    this.price = price;
  }

  @Override
  public boolean equals(Object o) {
    if (this == o) {
      return true;
    }
    if (!(o instanceof EmbeddedIdBean)) {
      return false;
    }
    EmbeddedIdBean that = (EmbeddedIdBean) o;
    return Objects.equals(name, that.name) && Objects.equals(intCode,
        that.intCode) && Objects.equals(price, that.price);
  }

  @Override
  public int hashCode() {
    return Objects.hash(name, intCode, price);
  }

  @Override
  public String toString() {
    return "EmbeddedIdBean{" +
        "name='" + name + '\'' +
        ", intCode=" + intCode +
        ", price=" + price +
        '}';
  }
}

File: web/src/test/kotlin/top/bettercode/summer/test/web/support/EmbeddedIdBean.kt
package top.bettercode.summer.test.web.support

import java.util.*

class EmbeddedIdBean {
    var name: String? = null
    var intCode: Int? = null
    var price: Long? = null
    override fun equals(other: Any?): Boolean {
        if (this === other) {
            return true
        }
        if (other !is EmbeddedIdBean) {
            return false
        }
        return name == other.name && intCode == other.intCode && price == other.price
    }

    override fun hashCode(): Int {
        return Objects.hash(name, intCode, price)
    }

    override fun toString(): String {
        return "EmbeddedIdBean{" +
                "name='" + name + '\'' +
                ", intCode=" + intCode +
                ", price=" + price +
                '}'
    }
}